Sources/NIOPosix/HappyEyeballs.swift

```diff
@@ -303,10 +304,43 @@ internal class HappyEyeballsConnector {
         self.connectionDelay = connectionDelay
     }
 
+    @inlinable
+    init(
+        resolver: Resolver,
+        loop: EventLoop,
+        host: String,
+        port: Int,
+        connectTimeout: TimeAmount,
+        resolutionDelay: TimeAmount = .milliseconds(50),
+        connectionDelay: TimeAmount = .milliseconds(250),
+        channelBuilderCallback: @escaping (EventLoop, NIOBSDSocket.ProtocolFamily) -> EventLoopFuture<Channel>
+    ) where ChannelBuilderResult == Void {
+        self.resolver = resolver
+        self.loop = loop
+        self.host = host
+        self.port = port
+        self.connectTimeout = connectTimeout
+        self.channelBuilderCallback = { eventLoop, protocolFamily in channelBuilderCallback(eventLoop, protocolFamily).map { ($0, ()) } }
+        self.resolutionTask = nil
+        self.connectionTask = nil
+        self.timeoutTask = nil
+
+        self.state = .idle
+        self.resolutionPromise = self.loop.makePromise()
+        self.error = NIOConnectionError(host: host, port: port)
+
+        precondition(resolutionDelay.nanoseconds > 0, "Resolution delay must be greater than zero, got \(resolutionDelay).")
+        self.resolutionDelay = resolutionDelay
+
+        precondition(connectionDelay >= .milliseconds(100) && connectionDelay <= .milliseconds(2000), "Connection delay must be between 100 and 2000 ms, got \(connectionDelay)")
+        self.connectionDelay = connectionDelay
+    }
```

Comment:
These two inits entirely duplicate their code, for one line of difference. Can we have one call the other?